src/main/java/hexgui/gui/AnalyzeDialog.java

```diff
@@ -0,0 +1,483 @@
+    JPanel leftPanel = new JPanel();
+    optionsPanel.add(leftPanel);
+    Box leftBox = Box.createVerticalBox();
+    leftPanel.add(leftBox);
+    m_autoRun = new JCheckBox("Autorun");
+    m_autoRun.addItemListener(
+        new ItemListener() {
+          public void itemStateChanged(ItemEvent e) {
+            if (!m_autoRun.isSelected()) m_listener.actionClearAnalyzeCommand();
+          }
+        });
+    m_autoRun.setToolTipText("Autorun");
+    m_autoRun.setEnabled(false);
+    leftBox.add(m_autoRun);
+    m_clearBoard = new JCheckBox("Clearboard");
+    m_clearBoard.setToolTipText("Clearboard");
+    m_clearBoard.setEnabled(false);
+    leftBox.add(m_clearBoard);
+    m_clearBoard.setSelected(true);
+    m_reuseWindow = new JCheckBox("Reuse Text");
+    m_reuseWindow.setToolTipText("Reuse text window");
+    leftBox.add(m_reuseWindow);
+    JPanel rightPanel = new JPanel();
+    rightPanel.add(createColorPanel());
+    optionsPanel.add(rightPanel);
+
+    // TODO: The following horizontal glue does not really work as expected
+    // (tested on Linux/Sun Java 1.6.0_14) and the left two components in
+    // the box are not aligned to the left.
+    optionsPanel.add(Box.createHorizontalGlue());
+
+    // TODO: If GTK Looks L&F is used on Linux/Sun Java 1.6.0_14 or OpenJDK
+    // 6b14-1.4.1-0ubuntu11, then the text of the checkbox items can be
+    // truncated a bit on the left (wrong minimum size calculation?). The
+    // two fillers are a workaround for this.
+    optionsPanel.add(GuiUtil.createFiller());
+    optionsPanel.add(GuiUtil.createFiller());
+
+    lowerPanel.add(createButtons());
+    m_comboBoxHistory.addActionListener(this);
+    return panel;
+  }
+
+  private String getComboBoxItem(int i) {
+    return m_comboBoxHistory.getItemAt(i).toString();
+  }
+
+  private int getComboBoxItemCount() {
+    return m_comboBoxHistory.getItemCount();
+  }
+
+  private int getCommandIndex(String label) {
+    for (int i = 0; i < m_commands.size(); ++i)
+      if (m_commands.get(i).getLabel().equals(label)) return i;
+    return -1;
+  }
+
+  private int getSelectedCommand() {
+    Object item = m_comboBoxHistory.getSelectedItem();
+    if (item == null) return -1;
+    return getCommandIndex(item.toString());
+  }
+
+  private void insertComboBoxItem(String label, int index) {
+    m_comboBoxHistory.insertItemAt(GuiUtil.createComboBoxItem(label), index);
+  }
+
+  private void loadRecent() {
+    m_comboBoxHistory.removeAllItems();
+    m_fullRecentList = PrefUtil.getList("net/sf/hexgui/gui/analyzedialog/recentcommands");
+    for (int i = 0; i < m_fullRecentList.size(); ++i) {
+      String name = m_fullRecentList.get(i);
+      if (getCommandIndex(name) >= 0) m_comboBoxHistory.addItem(GuiUtil.createComboBoxItem(name));
+      if (m_comboBoxHistory.getItemCount() > 20) break;
+    }
+    int index = getSelectedCommand();
+    if (index >= 0) selectCommand(index);
+    m_firstIsTemp = false;
+  }
+
+  private void runCommand() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.